prime-router/src/main/kotlin/config/validation/ConfigurationValidationService.kt

```diff
@@ -0,0 +1,67 @@
+package gov.cdc.prime.router.config.validation
+
+import java.io.File
+import java.io.InputStream
+
+/**
+ * Service that combines both the JSON schema and
+ * value validation into a single service call
+ */
+interface ConfigurationValidationService {
+
+    /**
+     * Validates YAML file structure and values
+     */
+    fun <T> validateYAML(
+        configType: ConfigurationType<T>,
+        file: File,
+    ): ConfigurationValidationResult<T>
+
+    /**
+     * Validates YAML string structure and values
+     */
+    fun <T> validateYAML(
+        configType: ConfigurationType<T>,
+        yamlString: String,
+    ): ConfigurationValidationResult<T>
+
+    /**
+     * Validates YAML stream structure and values
+     */
+    fun <T> validateYAML(
+        configType: ConfigurationType<T>,
+        inputStream: InputStream,
+    ): ConfigurationValidationResult<T>
+}
+
+class ConfigurationValidationServiceImpl(
+    private val jsonSchemaValidationService: JsonSchemaValidationService = JsonSchemaValidationServiceImpl(),
+    private val configurationValueValidationService: ConfigurationValueValidationService =
+    ConfigurationValueValidationServiceImpl(),
+) : ConfigurationValidationService {
+
+    override fun <T> validateYAML(configType: ConfigurationType<T>, file: File): ConfigurationValidationResult<T> {
```

Comment:
I overloaded it because I wanted to use some simple strings in the tests.

I also wanted to use some shared logic between both files and strings and found that input streams would let me handle both the same way.